ign_rviz_plugins/include/ignition/rviz/plugins/PoseWithCovarianceDisplay.hpp

```diff
@@ -0,0 +1,298 @@
+
+#ifndef IGNITION__RVIZ__PLUGINS__POSEWITHCOVARIANCEDISPLAY_HPP_
+#define IGNITION__RVIZ__PLUGINS__POSEWITHCOVARIANCEDISPLAY_HPP_
+
+#include <ignition/rendering.hh>
+
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+
+#include <QColor>
+
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "ignition/rviz/plugins/message_display_base.hpp"
+#include "ignition/rviz/plugins/CovarianceVisual.hpp"
+
+namespace ignition
+{
+namespace rviz
+{
+namespace plugins
+{
+/**
+ * @brief Helper for visualization PoseWithCovarianceStamped data as Arrow
+ */
+struct ArrowVisualPrivate
+{
+  /**
+   * @brief Update the arrow shaft and head visual length and radius
+   */
+  void updateVisual()
+  {
+    visual->Shaft()->SetLocalScale(shaftRadius * 2.0, shaftRadius * 2.0, shaftLength);
+    visual->SetOrigin(0, 0, -shaftLength);
+    visual->Head()->SetLocalScale(headRadius * 2.0, headRadius * 2.0, headLength * 2.0);
+  }
+
+  ignition::rendering::ArrowVisualPtr visual;
+  ignition::rendering::MaterialPtr mat;
+  float shaftLength = 1.0;
+  float shaftRadius = 0.05;
+  float headLength = 0.25;
+  float headRadius = 0.1;
+};
+
+/**
+ * @brief Helper for visualization PoseWithCovarianceStamped data as Axis
+ */
+struct AxisVisualPrivate
+{
+  /**
+   * @brief Update the axis visual length and radius
+   */
+  void updateVisual()
+  {
+    for (int i = 0; i < 3; ++i) {
+      auto arrow = std::dynamic_pointer_cast<rendering::ArrowVisual>(visual->ChildByIndex(i));
+      arrow->SetLocalScale(radius * 20, radius * 20, length * 2);
+    }
+  }
+
+  ignition::rendering::AxisVisualPtr visual;
+  float length = 1.0;
+  float radius = 0.1;
+  bool headVisible = false;
+};
+
+/**
+ * @brief Renders data from geometry_msgs::msg::PoseWithCovarianceStamped message as arrows or axes
+ */
+class PoseWithCovarianceDisplay : public MessageDisplay<geometry_msgs::msg::PoseWithCovarianceStamped>
+{
+  Q_OBJECT
+
+  /**
+   *  @brief Topic List
+   */
+  Q_PROPERTY(
+    QStringList topicList
+    READ getTopicList
+    NOTIFY topicListChanged
+  )
+
+public:
+  /**
+   * Constructor for PoseWithCovarianceStamped visualization plugin
+   */
+  PoseWithCovarianceDisplay();
+
+  // Destructor
+  ~PoseWithCovarianceDisplay();
+
+  // Documentation Inherited
+  void LoadConfig(const tinyxml2::XMLElement * /*_pluginElem*/) override;
+
+  // Documentation Inherited
+  void initialize(rclcpp::Node::SharedPtr _node) override;
+
+  // Documentation Inherited
+  void callback(const geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr _msg) override;
+
+  // Documentation inherited
+  void setTopic(const std::string & topic_name) override;
+
+  // Documentation inherited
+  void subscribe() override;
+
+  // Documentation inherited
+  void reset() override;
+
+  /**
+   * @brief Set ROS Subscriber topic through GUI
+   * @param[in] topic_name ROS Topic Name
+   */
+  Q_INVOKABLE void setTopic(const QString & topic_name);
```

Comment:
Follow the same naming convention
```suggestion
  /**
   * @brief Set ROS Subscriber topic through GUI
   * @param[in] _topic_name ROS Topic Name
   */
  Q_INVOKABLE void setTopic(const QString & _topic_name);
```